### PR TITLE
update fetch APIs to only process entities with deleted=false

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java [hunk 1]
```diff
@@ -65,7 +65,7 @@
 
   // Queries
   private static final String GET_DATASET_BY_IDS_QUERY =
-      "From DatasetEntity ds where ds.id IN (:ids)";
+      "From DatasetEntity ds where ds.id IN (:ids) AND ds.deleted = false ";
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java]: Should we use `ModelDBConstants.DELETED` here too?

File: backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java [hunk 2]
```diff
@@ -60,12 +60,20 @@
...
+  private static final String CHECK_DATASET_VERSION_EXISTS_BY_ID_HQL =
+      new StringBuilder("Select count(*) From DatasetVersionEntity dsv where ")
+          .append(" dsv." + ModelDBConstants.ID + " = :datasetVersionId ")
+          .append(" AND dsv." + ModelDBConstants.DELETED + " = false ")
           .toString();
   private static final String DATASET_VERSION_BY_DATA_SET_IDS_QUERY =
       "From DatasetVersionEntity ds where ds.dataset_id IN (:datasetIds) ";
   private static final String DATASET_VERSION_BY_IDS_QUERY =
-      "From DatasetVersionEntity ds where ds.id IN (:ids)";
+      "From DatasetVersionEntity ds where ds.id IN (:ids) AND ds.deleted = false ";
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java]: Should we use ModelDBConstants.DELETED here too?

File: backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java [hunk 3]
```diff
@@ -60,12 +60,20 @@
...
           .append(" = :datasetId AND dsv.")
           .append(ModelDBConstants.VERSION)
-          .append(" =:version")
+          .append(" =:version ")
+          .append(" AND dsv.")
+          .append(ModelDBConstants.DELETED)
+          .append(" = false ")
+          .toString();
+  private static final String CHECK_DATASET_VERSION_EXISTS_BY_ID_HQL =
+      new StringBuilder("Select count(*) From DatasetVersionEntity dsv where ")
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java]: Don't `count(*)`. Count on top of an index.

File: backend/src/main/java/ai/verta/modeldb/experiment/ExperimentDAORdbImpl.java [hunk 4]
```diff
@@ -84,7 +84,7 @@
           .append(" = :experimentId AND attr.field_type = :fieldType")
           .toString();
   private static final String EXPERIMENT_BY_BATCH_IDS_QUERY =
-      "From ExperimentEntity ex where ex.id IN (:ids)";
+      "From ExperimentEntity ex where ex.id IN (:ids) AND ex.deleted = false";
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experiment/ExperimentDAORdbImpl.java]: Should we use ModelDBConstants.DELETED here too?

File: backend/src/main/java/ai/verta/modeldb/experiment/ExperimentDAORdbImpl.java [hunk 5]
```diff
@@ -584,7 +584,7 @@ public Experiment getExperiment(List<KeyValue> keyValues) throws InvalidProtocol
           stringQueryBuilder.append(" AND ");
         }
       }
-      Query query = session.createQuery(stringQueryBuilder.toString());
+      Query query = session.createQuery(stringQueryBuilder.toString() + " AND ee.deleted = false ");
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experiment/ExperimentDAORdbImpl.java]: Should we use ModelDBConstants.DELETED here too?

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java [hunk 6]
```diff
@@ -110,13 +110,15 @@
...
+          .append(" AND ere." + ModelDBConstants.DELETED + " = false ")
           .toString();
   private static final String CHECK_EXP_RUN_EXISTS_AT_UPDATE_HQL =
       new StringBuilder("Select count(*) From ExperimentRunEntity ere where ")
           .append(" ere." + ModelDBConstants.ID + " = :experimentRunId ")
+          .append(" AND ere." + ModelDBConstants.DELETED + " = false ")
           .toString();
   private static final String GET_EXP_RUN_BY_IDS_HQL =
-      "From ExperimentRunEntity exr where exr.id IN (:ids)";
+      "From ExperimentRunEntity exr where exr.id IN (:ids) AND exr.deleted = false ";
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java]: Should we use ModelDBConstants.DELETED here too?

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java [hunk 7]
```diff
@@ -1714,7 +1720,7 @@ public ExperimentRun deepCopyExperimentRunForUser(
           stringQueryBuilder.append(" AND ");
         }
       }
-      Query query = session.createQuery(stringQueryBuilder.toString());
+      Query query = session.createQuery(stringQueryBuilder.toString() + " AND er.deleted = false ");
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java]: Should we use ModelDBConstants.DELETED here too?

File: backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java [hunk 8]
```diff
@@ -125,8 +125,10 @@
           .append(ModelDBConstants.PROJECT_IDS)
           .append(")")
           .toString();
-  private static final String GET_PROJECT_BY_ID_HQL = "From ProjectEntity p where p.id = :id";
-  private static final String GET_PROJECT_BY_IDS_HQL = "From ProjectEntity p where p.id IN (:ids)";
+  private static final String GET_PROJECT_BY_ID_HQL =
+      "From ProjectEntity p where p.id = :id AND p.deleted = false";
+  private static final String GET_PROJECT_BY_IDS_HQL =
+      "From ProjectEntity p where p.id IN (:ids) AND p.deleted = false";
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java]: Should we use ModelDBConstants.DELETED here too?

**Review by conradoverta — COMMENTED** [review]
1. lots of mixing between using `ModelDBConstants.DELETED` and not using
2. let's make sure we update/add indices to cover all the modified queries. It can be a follow up PR

**Comment by ravishetye** [comment]
no other comments than what @conradoverta  has mentioned already

File: backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java]: updated with constant name

File: backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java]: updated with constant name

File: backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java (hunk 3 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java]: I do not understand this comment. we have just got the count based on condition then what is the wrong here?

File: backend/src/main/java/ai/verta/modeldb/experiment/ExperimentDAORdbImpl.java (hunk 4 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experiment/ExperimentDAORdbImpl.java]: updated with constant name

File: backend/src/main/java/ai/verta/modeldb/experiment/ExperimentDAORdbImpl.java (hunk 5 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experiment/ExperimentDAORdbImpl.java]: updated with constant name

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java (hunk 6 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java]: updated with constant name

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java (hunk 7 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java]: updated with constant name

File: backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java (hunk 8 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java]: updated with constant name

File: backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java (hunk 3 above)
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java]: if you do a count(id) on the table and there is an index then there is high chance that tidb will pick the right index.

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java]: Ok make sense

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java]: fixed

**Review by ravishetye — APPROVED** [review]

File: backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java (hunk 3 above)
Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java]: We know `count(*)` is slow. We don't know why, but we get large processing time on hibernate doing that. We don't need to us to do it too if we have a reasonable alternative.

**Pull request merged**